apps/scan/backend/src/scanners/custom/state_machine.ts

```diff
@@ -0,0 +1,1303 @@
+                  cond: (context) => {
+                    assert(context.interpretation);
+                    return context.interpretation.type === 'InvalidSheet';
+                  },
+                },
+                {
+                  target: '#needs_review',
+                  cond: (context) => {
+                    assert(context.interpretation);
+                    return context.interpretation.type === 'NeedsReviewSheet';
+                  },
+                },
+              ],
+            },
+          },
+        },
+        ready_to_accept: {
+          id: 'ready_to_accept',
+          invoke: pollPaperStatus,
+          on: {
+            ACCEPT: 'accepting',
+            SCANNER_READY_TO_EJECT: doNothing,
+          },
+        },
+        accepting: acceptingState,
+        accepted: {
+          id: 'accepted',
+          entry: (context) => recordAcceptedSheet(workspace.store, context),
+          invoke: pollPaperStatus,
+          initial: 'scanning_paused',
+          on: { SCANNER_NO_PAPER: doNothing },
+          states: {
+            scanning_paused: {
+              on: {
+                SCANNER_READY_TO_SCAN: doNothing,
+                SCANNER_BOTH_SIDES_HAVE_PAPER: doNothing,
+                SCANNER_JAM: doNothing,
+                SCANNER_READY_TO_EJECT: doNothing,
+              },
+              after: {
+                DELAY_ACCEPTED_READY_FOR_NEXT_BALLOT:
+                  '#checking_paper_status_after_scan',
+              },
+            },
+          },
+          after: {
+            DELAY_ACCEPTED_RESET_TO_NO_PAPER:
+              '#checking_paper_status_after_scan',
+          },
+        },
+        needs_review: {
+          id: 'needs_review',
+          invoke: pollPaperStatus,
+          on: {
+            ACCEPT: 'accepting_after_review',
+            RETURN: 'returning',
+            SCANNER_READY_TO_EJECT: doNothing,
+          },
+        },
+        accepting_after_review: acceptingState,
+        returning_to_rescan: {
+          id: 'returning_to_rescan',
+          ...rejectingState('#no_paper'),
+        },
+        returning: { id: 'returning', ...rejectingState('#returned') },
+        returned: {
+          id: 'returned',
+          invoke: pollPaperStatus,
+          on: {
+            SCANNER_READY_TO_SCAN: doNothing,
+            SCANNER_NO_PAPER: 'no_paper',
+          },
+        },
+        rejecting: {
+          id: 'rejecting',
+          ...rejectingState('#rejected'),
+        },
+        // Paper has been rejected and is held in the front, waiting for removal.
+        rejected: {
+          id: 'rejected',
+          invoke: pollPaperStatus,
+          on: {
+            SCANNER_READY_TO_SCAN: doNothing,
+            SCANNER_NO_PAPER: 'no_paper',
+          },
+        },
+        // There is a jam that the custom scanner itself has notified us of. We have
+        // to reset the hardware in order to clear the jam when done.
+        internal_jam: {
```

Comment:
I'm honestly not sure anymore if we need both the internal_jam and jam states here but I'll make a note for myself to follow up and play with if we can simplify that